Когда используется критерий Стьюдента, а когда Z –критерий? 

### Критерий Стьюдента:

Используется, когда известны только среднее значение и стандартное отклонение выборки, а размер выборки мал (<30).
Применяется для проверки гипотез о равенстве средних значений двух независимых выборок (например, сравнение средних результатов двух групп испытуемых).
Также может быть использован для проверки гипотезы о равенстве среднего значения выборки некоторому фиксированному значению.

### Z-критерий:

Используется, когда известны среднее значение и стандартное отклонение генеральной совокупности, а размер выборки большой (нормальное распределение).
Применяется для проверки гипотез о равенстве средних значений двух выборок, когда известны параметры генеральных совокупностей.
Обычно используется, когда размер выборки больше 30. При этом, если размер выборки меньше, но известна дисперсия генеральной совокупности, также можно использовать Z-критерий.

### Задача 2

Проведите тест гипотезы. Утверждается, что шарики для подшипников, изготовленные автоматическим станком, имеют средний диаметр 17 мм. Используя односторонний критерий с α=0,05, проверить эту гипотезу, если в выборке из n=100 шариков средний диаметр оказался равным 17.5 мм, а дисперсия известна и равна 4 кв. мм.

mu = 17

X = 17.5

n = 100

s = 4**0.5 = 2

α=0,05

используем формулу для Z критерия $$Z = \frac{(X̄ - μ)}{\frac{σ}{\sqrt{n}}}$$

$$Z = \frac{(17.5 - 17)}{\frac{2}{\sqrt{100}}}$$

$$Z = 2.5$$

Воспоьлзуемся таблицей нормального Z распределения

![Таблица нормального Z распределения](https://znachek.pro/uploads/posts/2023-05/1685507208_znachek-pro-p-tablitsa-normalnogo-raspredeleniya-veroyat-11.jpg)

$$Z_крит = Z_{1 - α} = Z_{0.95} ≈ 1.645$$

Поскольку 2.5 > 1.645, у нас есть достаточные основания отклонить нулевую гипотезу на уровне значимости 0,05.


### Задача 3

Проведите тест гипотезы. Продавец утверждает, что средний вес пачки печенья составляет 200 г. Из партии извлечена выборка из 10 пачек. Вес каждой пачки составляет: 202, 203, 199, 197, 195, 201, 200, 204, 194, 190. Известно, что их веса распределены нормально. Верно ли утверждение продавца, если учитывать, что доверительная вероятность равна 99%? (Провести двусторонний тест.)

Поскольку дисперсия генеральной совокупности неизвестна испоьлзуем t-распределение

mu = 200

avg = (202 + 203 + 199 + 197 + 195 + 201 + 200 + 204 + 194 + 190) / 10 = 198.5г

s = √[((202 - 198.5)² + (203 - 198.5)² + ... + (190 - 198.5)²) / (10 - 1)] ≈ 4.16

Используем формулу $$ t = \frac{avg-\mu}{\frac{\sigma}{\sqrt{n}}} $$

$$ t = \frac{198.5-200}{\frac{4.16}{\sqrt{10}}} = -1.14$$

Оприделим критиеские значение  для критерия alfa = 1-0.99 = 0.01 по таблице при степенях свободы 10-1 = 9

![Критические значения критерия Стьюдента](https://cf2.ppt-online.org/files2/slide/6/6AVb07fhBxo9I3mSvpjgwTcFJOtKHd5asWRnUr8kM/slide-3.jpg)

$$t_{крит} = t_{0.01} = 3.25$$

Поскольку t = -1.14 входит в диапазон (-3.25, 3.25] то утверждение продавца верное

### Задача 4

Есть ли статистически значимые различия в росте дочерей? Рост матерей 172, 177, 158, 170, 178,175, 164, 160, 169, 165 Рост взрослых дочерей: 173, 175, 162, 174, 175, 168, 155, 170, 160

In [18]:
import numpy as np
from scipy import stats

mother_heights = np.array([172, 177, 158, 170, 178, 175, 164, 160, 169, 165])
daughter_heights = np.array([173, 175, 162, 174, 175, 168, 155, 170, 160])

# stats.ttest_rel(daughter_heights, mother_heights[:-1])       #ttest_rel(daughter_heights, mother_heights)

t_statistic, p_value = stats.ttest_rel(daughter_heights, mother_heights[:-1])

print("t-статистика:", t_statistic)
print("p-value:", p_value)

print('Возьмем статистическую значимость 5%. \nПолученное значение p-value больше стандартных значений статистической значимости (1,5,10%)) 0.59 > (0.01, 0.05, 0.1) \nПоэтому статистических различий у данных групп нет')

t-статистика: -0.559522990335608
p-value: 0.5911212354055175
Возьмем статистическую значимость 5%. 
Полученное значение p-value больше стандартных значений статистической значимости (1,5,10%)) 0.59 > (0.01, 0.05, 0.1) 
Поэтому статистических различий у данных групп нет


In [19]:
print(stats.t.ppf(0.975, 19-1))

2.10092204024096


Поскольку результат теста -0.56 а табличный критерий для  2,5% составляет -2,1 до 2,1 то наши значения попадают в диапазон нулевой гипотезы и статистических различий нет